inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -245,7 +245,7 @@ def revert(self, rebooter: Rebooter, time_to_wait_before_reboot: int) -> None:
         """
         logger.debug("")
         dispatcher_state.clear_dispatcher_state()
-        if self.snap_num:
+        if self.snap_num and self.snap_num is not "0":
             self._rollback_and_delete_snap()
             time.sleep(time_to_wait_before_reboot)
             rebooter.reboot()
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Maintainability$</h2>**
**Suggestion:** Use the equality operator '!=' for string comparison to maintain consistency and readability.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if self.snap_num and self.snap_num != "0":
            self._rollback_and_delete_snap()
            time.sleep(time_to_wait_before_reboot)
            rebooter.reboot()
```

Reply:
Agreed. Using  '!=' directly for string comparison is more idiomatic and clearer